core/src/main/java/com/linecorp/armeria/server/ExceptionReportingServerErrorHandler.java

```diff
@@ -85,7 +85,7 @@ public HttpResponse onServiceException(ServiceRequestContext ctx, Throwable caus
                 cause.getCause() != null;
 
         if (ctx.shouldReportUnhandledExceptions() && !isExpectedException) {
-            if (scheduledUpdater.compareAndSet(this, 0, 1)) {
```

Comment:
You don't actually need this optimization because `compareAndSet()` does it for you.